streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -946,14 +934,14 @@ private void completeShutdown(final boolean cleanRun) {
         log.info("Shutting down");
 
         try {
-            taskManager.shutdown(cleanRun);
+            restoreThread.shutdown(10_000L);
```

Comment:
Why just choose a random number here...? Shouldn't we apply the timeout supplied to `KafkaStreams#close`?